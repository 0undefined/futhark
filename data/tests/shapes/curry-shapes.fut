// Test that shape declarations are taken into account even when the
// function is curried.
//
// At the time this test was written, the only way to determine the
// success of this is to inspect the result of internalisation by the
// compiler.

fun [int,!n] oneToEach([int,n] r) =
  map(+1, r)

fun [[int]] main([[int]] a) =
  map(oneToEach, a)
