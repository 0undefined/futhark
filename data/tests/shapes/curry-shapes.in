[[6,5,2,1],
 [4,5,9,-1]]
