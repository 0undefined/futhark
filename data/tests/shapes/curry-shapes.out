[[7,6,3,2],
 [5,6,10,0]]
